backend/uclapi/workspaces/occupeye.py

```diff
@@ -791,6 +803,103 @@ def get_survey_sensors_summary(self, survey_ids):
 
         return sensors_data_dict.values()
 
+    def _cache_historical_time_usage_data(self, survey_id, day_count):
+        """
+        Function to cache in Redis the historical usage data over each 10
+        minute time period.
+        day_count is the number of days of historical data (from including
+        yesterday, but not including today) to cache.
+        E.g. if you cache on Tuesday for 7 days, all the data from
+        last Sunday to Monday (e.g. yesterday) inclusive will be cached.
+        This is to support apps which show historical survey usage data.
+        """
+        end_date = datetime.now() - timedelta(days=1)
+        start_date = end_date - timedelta(days=day_count - 1)
+        headers = {
+            "Authorization": self.get_bearer_token()
+        }
+
+        url = (
+            "{}/api/Query?"
+            "Deployment={}&"
+            "startdate={}&"
+            "enddate={}&"
+            "SurveyID={}&"
+            "QueryType=ByDateTime&"
+            "StartTime=00%3A00&"
+            "EndTime=24%3A00&"
+            "GroupBy[]=TriggerDate&"
+            "GroupBy[]=TimeSlot&"
+            "WeekDays[]=Monday&"
+            "WeekDays[]=Tuesday&"
+            "WeekDays[]=Wednesday&"
+            "WeekDays[]=Thursday&"
+            "WeekDays[]=Friday&"
+        ).format(
+            self.base_url,
+            self.deployment_name,
+            start_date.strftime("%Y-%m-%d"),
+            end_date.strftime("%Y-%m-%d"),
+            survey_id
+        )
+
+        request = requests.get(url, headers=headers)

```

Comment:
or maybe like this? https://stackoverflow.com/a/17788913/825916

Reply:
Where does OAuth come into this? Or did you mean with `params=params`?